### PR TITLE
Fix udev.resolve_glob to match device path too

### PR DESCRIPTION
**Pull request by vojtechtrefny**
Related: rhbz#1200833

-------
Much simpler version of #441

### CODE REVIEW

**Comment by vpodzime** [comment]
Looks good to me.

**Comment by vojtechtrefny** [comment]
Update: New commit -- added `disks_only` argument to `resolve_glob` for disk-only results.

**Comment by dwlehman** [comment]
What is the disks_only argument for? It seems like a strange argument for this particular function.

**Comment by vojtechtrefny** [comment]
The `resolve_glob` is used in Anaconda during parsing kickstart options like `ignored-disks` etc. so it makes sense to match just disks. They don't have devicetree there and it's quite complicated to filter disks using just the name and udev.

**Comment by M4rtinK** [comment]
>The resolve_glob is used in Anaconda during parsing kickstart options like ignored-disks etc. so it makes sense to match just disks. They don't have devicetree there and it's quite complicated to filter disks using just the name and udev.

Yep, there is no devicetree in the kickstart parsing phase, but we do need to know if a device specification is a disk or not for `clearpart` or `ignoredisk`.

**Comment by vojtechtrefny** [comment]
Added a new method to get the udev device/info from name.

File: blivet/udev.py [hunk 1]
```diff
@@ -47,6 +47,15 @@ def get_device(sysfs_path):
 
     return dev
 
+def get_device_by_name(name, subsystem="block"):
+    try:
+        dev = pyudev.Device.from_name(global_udev, subsystem, name)
```
Comment (dwlehman) [inline blivet/udev.py]: Note that this will only match "system" names, eg: `md0`, `dm-1`, `sda`.

**Comment by dwlehman** [comment]
I hate to rain on your parade, but...

I'm wondering if it would make more sense for anaconda to do a combination of `resolve_glob` and filtering based on `device_is_disk`, which leads me to point out that `udev.device_is_disk` is woefully inadequate for your purposes here. Any device-mapper device will get a `True` from that function. `Populator._udev_device_is_disk` has what you're after, but I think you'd have to move it into `blivet.udev` for this all to work.

**Comment by vojtechtrefny** [comment]
@dwlehman: Would it make more sense to change `udev.device_is_disk` to make all the checks that `Populator._udev_device_is_disk` does (probably just on master, not for rhel7-branch, because it's an API change)? We use it (`udev.device_is_disk`) only from populator and populator.helpers.disk and we  always filter out that "non-disk" matches. Or is a separate method (in udev.py) better?

**Comment by dwlehman** [comment]
That definitely seems like the right thing to do (move `Populator._udev_device_is_disk` to `udev.device_is_disk`), and I think it will actually work.

**Comment by vojtechtrefny** [comment]
Added a new function `udev.device_is_realdisk` that will do all the checks `Populator._udev_device_is_disk` does on master. I'll create a separate PR for master/3.0-branch that will change behaviour `udev.device_is_disk` there.

**Comment by M4rtinK** [comment]
The udev.device_is_realdisk() function works fine for my usecase (https://github.com/rhinstaller/anaconda/pull/682), thanks! :)

File: blivet/udev.py [hunk 2]
```diff
@@ -286,6 +286,29 @@ def device_is_loop(info):
     return (device_get_name(info).startswith("loop") and
             os.path.isdir("%s/loop" % device_get_sysfs_path(info)))
 
+def device_is_realdisk(info):
```
Comment (dwlehman) [inline blivet/udev.py]: You should change `DeviceTree.udevDeviceIsDisk` to call this function as well so the code isn't in two places.

**Comment by dwlehman** [comment]
Looks good otherwise.

File: blivet/udev.py (hunk 2 above)
Comment (vojtechtrefny) [inline blivet/udev.py]: Fixed.

**Comment by vpodzime** [comment]
Looks good to me.

**Pull request merged**